api/classes/base_controller.js

```diff
@@ -1,87 +1,70 @@
 var Boom = require('boom');
+var Promise = require('bluebird');
+var errors = require('./errors');
 
 function BaseController(model)  {
   this.Model = model;
 }
 
-BaseController.prototype.getAll = function(req, reply) {
-  var result = this.Model.fetchAll()
-  .then(function(records) {
-    if (!records) { throw Boom.notFound(); }
-    return req.generateResponse(records.toJSON())
-      .code(200);
-  });
-  return reply(result);
+BaseController.prototype.data = function(req) {
+  // Abstract base method,
+  // formats data for database entry
 };
 
 BaseController.prototype.getOne = function(req, reply) {
-  var result = this.Model.query({
-    where: {
-      id: req.params.id
-    }
-  }).fetch()
-  .then(function(record) {
-    if (!record) { throw Boom.notFound(); }
-    return req.generateResponse(record.toJSON())
-      .code(200);
-  });
-  return reply(result);
+  var record = this.pre.records[0];
```

Comment:
I don't know if this is a great pattern. I assume `pre` is from the Hapi route config? I don't think you should assume that that is where you are using it.